pkg/resources/kafka/kafka.go

```diff
@@ -1161,7 +1161,15 @@ func (r *Reconciler) createExternalListenerStatuses(log logr.Logger) (map[string
 						portNumber = eListener.ContainerPort
 					}
 					if brokerHost == "" {
-						brokerHost = bConfig.NodePortExternalIP[eListener.Name]
+						if bConfig.NodePortExternalIP[eListener.Name] == "" {
+							brokerHost, err = r.getK8sAssignedNodeAddress(broker.Id, string(bConfig.NodePortNodeAddressType))
+							if err != nil {
+								log.Error(err, fmt.Sprintf("could not get the (%s) address of the broker's (ID: %d) node for external listener (%s) configuration",
+									bConfig.NodePortNodeAddressType, broker.Id, eListener.Name))
+							}
+						} else {
+							brokerHost = bConfig.NodePortExternalIP[eListener.Name]
+						}
 					} else {
 						brokerHost = fmt.Sprintf("%s-%d-%s.%s%s", r.KafkaCluster.Name, broker.Id, eListener.Name, r.KafkaCluster.Namespace, brokerHost)
 					}
```

Comment:
**Suggestion**: I'd prefer to wrap all these if-else clauses with a util function, for example:
```golang
broekrHost := getBrokerHost(host)
```

The reason behind this is that the function `createExternalListenerStatuses` is already very cumbersome and difficult for others to understand all the logics in the function. Wrapping the if-else clauses would help improve the readability of this `createExternalListenerStatuses` function and make it easier for others to maintain in the future